### PR TITLE
Feature request #244 HTML report generation. Added --format which out…

### PR DESCRIPTION
**Pull request by GobiasSomeCoffeeCo**
…puts CSV for FeroxResponse. I'm sure it can be iterated on, but please let me know what you think and I will try to fix/improve it. 

# Landing a Pull Request (PR)

Long form explanations of most of the items below can be found in the [CONTRIBUTING](https://github.com/epi052/feroxbuster/blob/master/CONTRIBUTING.md) guide.

## Branching checklist
- [x] There is an issue associated with your PR (bug, feature, etc.. if not, create one)
- [x] Your PR description references the associated issue (i.e. fixes #123456)
- [x] Code is in its own branch
- [x] Branch name is related to the PR contents
- [x] PR targets main

## Static analysis checks
- [x] All rust files are formatted using `cargo fmt`
- [x] All `clippy` checks pass when running `cargo clippy --all-targets --all-features -- -D warnings -A clippy::mutex-atomic`
- [ ] All existing tests pass

## Documentation
- [ ] New code is documented using [doc comments](https://doc.rust-lang.org/stable/rust-by-example/meta/doc.html)
- [ ] Documentation about your PR is included in the README, as needed

## Additional Tests
- [x] New code is unit tested
- [ ] New code is integration tested, as needed
- [x] New tests pass

### CODE REVIEW

**Comment by epi052** [comment]
alright, sorry for the delay, but i wanted to get the other PR's across the finish line before this one. The first thing we'll need to do is merge main back into this branch. 

Can you let me know where the CSV comes into play? Is that an additional format you want to support, or is it to facilitate writing the html?

**Comment by epi052** [comment]
this looks like a good place to start with the html template. It's searchable, the columns are sortable, and it's not awful to look at :grin:  

https://codepen.io/heiswayi/pen/VvpmaE

File: src/config/container.rs [hunk 1]
```diff
@@ -610,6 +624,20 @@ impl Configuration {
...
 
+        if let Some(arg) = args.values_of("format") {
+            let argument = arg.map(|val| val.to_string()).collect();
+            if format_arg_check(&argument) {
+                config.format = argument
+            } else {
+                config.format = "Unsupported Format Type".to_string();
+            }
+            match config.format.as_str() {
+                "csv" => config.output_format = OutputFormat::Csv,
```
[the remark this answers is not in the document]
Comment (GobiasSomeCoffeeCo) [reply src/config/container.rs:635]: Hey @epi052! Finally got some time this weekend and implemented all of the changes, however, I'm running into a nasty little bug that I can't seem to figure out. For some reason whenever `-o` or `--output` is invoked it is now required that `--format` or `-F` (the short arg I went with). I keep getting a generic error `error: The argument 'format' wasn't found
`. Any idea where I'm going wrong? Thanks!

[the remark this answers is not in the document]
Comment (epi052) [reply src/config/container.rs:635]: not sure without looking; can you push a commit with your latest changes?

[the remark this answers is not in the document]
Comment (GobiasSomeCoffeeCo) [reply src/config/container.rs]: Pushed the commit. Not sure where I'm going wrong? Getting failed tests with the banner and also if I try and run anything with `--output` without giving the `--format` arg as well.

[the remark this answers is not in the document]
Comment (epi052) [reply src/config/container.rs]: hey there, got sidetracked. I'll take a look at this this evening/tomorrow

[the remark this answers is not in the document]
Comment (epi052) [reply src/config/container.rs]: got your branch working, going to go through here in a bit and reduce it to the bare minimum change to get to a working state, then i'll respond here with the fix

[the remark this answers is not in the document]
Comment (epi052) [reply src/config/container.rs]: looks like all you need to update is `config/container.rs`
```diff
-        config.format = value_t!(args, "format", OutputFormat).unwrap_or_else(|e| e.exit());
-
+       update_config_if_present!(&mut config.format, args, "format", OutputFormat);
```

**Pull request closed without merging**